plugin/PDFView.qml

```diff
@@ -349,7 +349,13 @@ SilicaFlickable {
             property int currentIndex
 
             model: pdfCanvas.document.searchModel
-            onModelChanged: moveToSearchMatch(0)
+            onModelChanged: if (model) model.onCountChanged.connect(moveToFirstMatch)
```

Comment:
I agree, currentIndex is only used in the root item functions, so I've moved it there and renamed it searchIndex.

But, I can't find how to move the moveToFirstMatch() stuff to the root item. I've done something like this:
```
Connections {
   target: searchDisplay
   onModelChanged: if (model) model.onCountChanged.connect(movetoFirstMatch))
}
function moveToFirstMatch() {
   if (!searchDisplay.count) return
   moveToSearchMatch(0)
   model.onCiountChanged.disconnect(moveToFirstMatch)
}
```
I'm stuck on the last line of the function. How to disconnect the signal handler for it to run only once (precisely, when the first found match is available) ?